lib/dom/xml_entities.nit

```diff
@@ -0,0 +1,586 @@
+
+	redef fun to_s do return "<![CDATA[{content}]]>"
+end
+
+# A Tag is a node in a DOM tree
+abstract class XMLTag
+	super XMLEntity
+
+	# The name of the tag
+	var tag_name: String
+end
+
+# Any kind of XML tag with attributes
+abstract class XMLAttrTag
+	super XMLTag
+
+	# List of attributes in a Tag
+	var attributes: Array[XMLAttribute]
+end
+
+# One-liner XML Tag (Ends with />)
+class XMLOnelinerTag
+	super XMLAttrTag
+
+	redef fun to_s do
+		var s = "<{tag_name}"
+		if not attributes.is_empty then
+			s += " "
+			s += attributes.join(" ")
+		end
+		s += "/>"
+		return s
+	end
+end
+
+# A (potentially) multi-line spanning XML Tag start
+class XMLStartTag
+	super XMLAttrTag
+
+	# Optional matching tag, must be matched for the document to be well-formed
+	var matching: nullable XMLEndTag
+
+	redef fun to_s do
+		var s = "<{tag_name}"
+		if not attributes.is_empty then
+			s += " "
+			s += attributes.join(" ")
+		end
+		s += ">"
+		for i in children do s += i.to_s
+		if matching != null then s += matching.to_s
+		return s
+	end
+end
+
+# Any prolog style-Tag (starting with <?xml)
+class XMLPrologTag
+	super XMLAttrTag
+
+	redef fun to_s do return """<?{{{tag_name}}} {{{attributes.join(" ")}}}?>"""
+end
+
+class XMLProcessingInstructionTag
+	super XMLTag
+
+	var content: String
+
+	redef fun to_s do return "<?{tag_name} {content}?>"
+end
+
+# An end Tag (starting with </)
+class XMLEndTag
+	super XMLTag
+
+	# Optional matching tag, must be matched for the document to be well-formed
+	var matching: nullable XMLStartTag
+
+	redef fun to_s do return "</{tag_name}>"
+end
+
+# An XML comment tag
+class XMLCommentTag
+	super XMLTag
+
+	redef fun to_s do return "<!--{tag_name}-->"
+end
+
+# A DOCTYPE Tag
+class XMLDoctypeTag
+	super XMLTag
+
+	var content: String
+
+	redef fun to_s do return "<!DOCTYPE {content}>"
+end
+
+# A Special Tag (starts with !)
+#
+# TODO: Support the remaining ! tags
+class XMLSpecialTag
+	super XMLTag
+
+	redef fun to_s do return "<!{tag_name}>"
+end
+
+# Attributes are contained in tags, they provide meta-information on a tag
+abstract class XMLAttribute
+	super XMLEntity
+
+	# Name of the attribute
+	var name: String
+end
+
+# An attribute with a String value
+class XMLStringAttr
+	super XMLAttribute
+
+	# Value of the attribute without the double quotes
+	var value: String
+
+	# Type of delimiter (can be either " or ')
+	var delimiter: Char
+
+	redef fun to_s do return "{name}={delimiter}{value}{delimiter}"
+end
+
+# Badly formed XML attribute
+class BadXMLAttribute
+	super XMLAttribute
+
+	redef fun to_s do return name
+end
+
+# Internal use only, shows the end of an attribute block
+private class XMLAttributeEnd
+	super XMLAttribute
+end
+
+# Any XML Error that happens when parsing
+class XMLError
+	super XMLEntity
+
+	# Error message reported by the parser
+	var message: String
+
+	redef fun to_s do return "XML Error: {message} at {location.to_s}"
+end
+
+# Provides XML parsing facilities
+class XMLProcessor
```

Comment:
I suggest that you move XMLProcessor and related things to its own module `xml_processor`